core/storage/src/lib.rs

```diff
@@ -966,6 +1018,119 @@ impl<Adapter: StorageAdapter> IbcCrossChainStorage for ImplStorage<Adapter> {
+            }
+            prefix
+                if prefix.starts_with(
+                    &<String as TryInto<LocalIbcPath>>::try_into(String::from("connections"))
+                        .unwrap(),
+                ) =>
+            {
+                impl_get_keys_by_schemas_for!(
+                    self,
+                    all_vec,
+                    &key_prefix,
+                    local_ibc_prefix,
+                    ConnectionEndSchema
+                );
+            }
+            prefix
+                if prefix.starts_with(
+                    &<String as TryInto<LocalIbcPath>>::try_into(String::from("channelEnds"))
+                        .unwrap(),
+                ) =>
+            {
+                impl_get_keys_by_schemas_for!(
+                    self,
+                    all_vec,
+                    &key_prefix,
+                    local_ibc_prefix,
+                    ChannelEndSchema
+                );
+            }
+            prefix
+                if prefix.starts_with(
+                    &<String as TryInto<LocalIbcPath>>::try_into(String::from("nextSequence"))
+                        .unwrap(),
+                ) =>
+            {
+                impl_get_keys_by_schemas_for!(
+                    self,
+                    all_vec,
+                    &key_prefix,
+                    local_ibc_prefix,
+                    SeqSendsSchema,
+                    SeqRecvsSchema,
+                    SeqAcksSchema
+                );
+            }
+            prefix
+                if prefix.starts_with(
+                    &<String as TryInto<LocalIbcPath>>::try_into(String::from("commitments"))
+                        .unwrap(),
+                ) =>
+            {
+                impl_get_keys_by_schemas_for!(
+                    self,
+                    all_vec,
+                    &key_prefix,
+                    local_ibc_prefix,
+                    PacketCommitmentSchema
+                );
+            }
+            prefix
+                if prefix.starts_with(
+                    &<String as TryInto<LocalIbcPath>>::try_into(String::from("acks")).unwrap(),
+                ) =>
+            {
+                impl_get_keys_by_schemas_for!(
+                    self,
+                    all_vec,
+                    &key_prefix,
+                    local_ibc_prefix,
+                    AcknowledgementCommitmentSchema
+                );
+            }
+            prefix
+                if prefix.starts_with(
+                    &<String as TryInto<LocalIbcPath>>::try_into(String::from("receipts")).unwrap(),
+                ) =>
+            {
+                impl_get_keys_by_schemas_for!(
+                    self,
+                    all_vec,
+                    &key_prefix,
+                    local_ibc_prefix,
+                    IbcReceiptSchema
+                );
+            }
+            _ => {}
+        }
+
+        Ok(all_vec)
+    }
```

Comment:
I think use [`regex`](https://docs.rs/regex/1.6.0/regex/) with `lazy_static` is better.